File: 0540-single-element-in-a-sorted-array/0540-single-element-in-a-sorted-array.go
func singleNonDuplicate(nums []int) int {
    l := 0
    r := len(nums) - 1
    
    for l < r {
        m := l + (r - l)/2
        mid := nums[m]
        var mid_cmp int
        
        if m % 2 == 0 {
            mid_cmp = nums[m + 1]
        } else {
            mid_cmp = nums[m - 1]
        }
        
        if mid == mid_cmp {
            l = m + 1
        } else {
            r = m
        }
    }
    
    return nums[l]
}

File: 0540-single-element-in-a-sorted-array/0540-single-element-in-a-sorted-array.ts
function singleNonDuplicate(nums: number[]): number {
    let l = 0,
        r = nums.length - 1
    
    while (l < r) {
        let m = Math.floor((l + r) / 2)
        let num = nums[m]
        let num_cmp = m % 2 === 0 ? nums[m + 1]:nums[m - 1]
        
        if (num === num_cmp) 
            l = m + 1
        else
            r = m
    }
    
    return nums[l]
};